File: luhn_algorithm.py
import sys

if len(sys.argv) != 2:
	print('Usage: python luhn_algorithm.py <number>')
	sys.exit(1)

number = sys.argv[1]

# Converting input to a list
try:
	number_list = [int(i) for i in number]
except ValueError:
	print('Usage: python luhn_algorithm.py <number>')
	sys.exit(2)

# Doubling numbers in odd positions
for i in range(0, len(number_list), 2):
	number_list[i] *= 2

# Adding two digit numbers
temp_list = []
for i in range(len(number_list)):
	if number_list[i] > 9:
		double_number_string = str(number_list[i])
		for j in double_number_string:
			temp_list.append(int(j))
		number_list[i] = sum(temp_list)
		temp_list.clear()

# Checking if the sum of numbers in the list is divisible by 10
if sum(number_list) % 10 == 0:
	print('Valid')
else:
	print('Invalid')

File: luhn_algorithm/luhn_algorithm.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_DIGITS 100

int main (int argc, char *argv[]) {

	// Checking if multiple arguments have been passed
	if (argc != 2) {
		printf("Usage: %s <number>\n", argv[0]);
		return 1;
	}

	char *input = argv[1];
	int length = strlen(input);
	int num[MAX_DIGITS];
	int sum;

	// Converting to int and throwing error if failed
	for (int i = 0; i < length; i++) {
		if (input[i] >= '0' && input[i] <= '9') {
			num[i] = input[i] - '0';
		} else {
			printf("Usage: %s <number>\n", argv[0]);
			return 2;
		}
	}

	// Doubling the number and checking if larger than 9
	for (int i = 0; i < 16; i += 2) {
		num[i] *= 2;
		if (num[i] > 9)
		{
			num[i] -= 9;
		}
	}

	// Adding all numbers of array 
	for (int i = 0; i < 16; i++) {
		sum += num[i];
	}

	// Checking if the sum of numbers in the array is divisible by 10
	if (sum % 10 == 0) {
		printf("Valid\n");
	} else {
		printf("Invalid\n");
	}

	return 0;
}